org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/ClientPreferences.java

```diff
@@ -195,6 +195,10 @@ public boolean isWorkspaceApplyEditSupported() {
 		return capabilities.getWorkspace() != null && isTrue(capabilities.getWorkspace().getApplyEdit());
 	}
 
+	public boolean isWorkspaceSnippetEditSupported() {
+		return Boolean.parseBoolean(extendedClientCapabilities.getOrDefault("snippetEditSupport", "false").toString());
```

Comment:
Would it be possible to use the standard `workspaceEdit.snippetEditSupport` capability here to make this work out of the box for clients supporting the 3.18 spec addition?

Reply:
Yes, but it needs to be added to https://github.com/eclipse-lsp4j/lsp4j/blob/main/org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend#L59 . @hopehadfield started this at https://github.com/eclipse-lsp4j/lsp4j/pull/847 .